var/spack/repos/builtin/packages/steps/package.py

```diff
@@ -0,0 +1,108 @@
+import os.path as osp
+
+from spack import *
+
+
+class Steps(CMakePackage):
+    """STochastic Engine for Pathway Simulation"""
+
+    homepage = "http://www.example.com"
+    git      = "https://github.com/CNS-OIST/STEPS.git"
+
+    version("3.3.0", submodules=True)
+    version("3.2.0", submodules=True)
+    version("develop", branch="master", submodules=True)
+
+    variant("native", default=True, description="Generate non-portable arch-specific code")
+    variant("bindings", default="cython",
+            values=("cython", "swig"),
+            description="Generate bindings using Cython, instead of Swig")
+    variant("lapack", default=False, description="Use new BDSystem/Lapack code for E-Field solver")
+    variant("petsc", default=False, description="Use PETSc library for parallel E-Field solver")
+    variant("mpi", default=True, description="Use MPI for parallel solvers")
+
+    depends_on("py-cython", when="bindings=cython")
+    depends_on("swig", when="bindings=swig")
+    depends_on("lapack", when="+lapack")
+    
+    depends_on("petsc^debug+int64", when="+petsc")
+    # @pramodk: I have spec conflict when I use python 3: spack spec steps%gcc+petsc^python@3
+    # because petsc still uses Python 2
+    # I tried to replace line above by:
+    #
+    # depends_on("petsc^debug+int64^python@3", when="+petsc^python3")
+    # depends_on("petsc^debug+int64^python@2", when="+petsc^python2")
+    #
+    # but `when` directive is always evaluate to False, whether I put python@2 or python@3
+    # please help!
+
+    depends_on("mpi", when="+mpi")
+    depends_on("blas")
+
+    def cmake_args(self):
+        args = []
+        spec = self.spec
+        bindings =  spec.variants["bindings"].value
+
+        if "+native" in spec:
+            args.append("-DTARGET_NATIVE_ARCH:BOOL=True")
+        else:
+            args.append("-DTARGET_NATIVE_ARCH:BOOL=False")
+
+        if bindings == "cython":
+            args.append("-DPYTHON_BINDINGS_CYTHON:BOOL=True")
+        elif bindings == "swig":
+            args.append("-DPYTHON_BINDINGS_CYTHON:BOOL=False")
+        else:
+            raise Exception("Unexpected bindings type: " + bindings)
+
+        if "+lapack" in spec:
+            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=True")
+        else:
+            args.append("-DUSE_BDSYSTEM_LAPACK:BOOL=False")
+
+        if "+petsc" in spec:
+            args.append("-DUSE_PETSC:BOOL=True")
+        else:
+            args.append("-DUSE_PETSC:BOOL=False")
+
+        if "+mpi" in spec:
+            args.append("-DUSE_MPI:BOOL=True")
+        else:
+            args.append("-DUSE_MPI:BOOL=False")
+
+        args.append('-DBLAS_LIBRARIES=' + spec['blas'].libs.joined(";"))
+
+        prefix = osp.join(
+            super(Steps, self).prefix,
+            'lib',
+            'python' + str(self.spec['python'].version.up_to(2)),
+            'site-packages'
+        )
+
+        args.append('-DPYTHON_INSTALL_PREFIX:PATH=' + prefix)
```

Comment:
As far as I have seen, people don't install python stuff like this (in python prefix). Two ways : 
- install as normal package and in `setup_environment` method append path to `PYTHONPATH` (this will get propagated into module)
- for pure python packages (not the case with steps), people use extensions / `spack activate` feature ([docs](https://spack.readthedocs.io/en/latest/packaging_guide.html#extensions))